v3/src/components/case-tile-common/inspector-panel/ruler-menu-list.tsx

```diff
@@ -61,14 +74,42 @@ export const RulerMenuList = observer(function RulerMenuList() {
       itemKey: "DG.Inspector.exportCaseData"
     },
     {
-      itemKey: "DG.Inspector.copyCaseDataToClipboard"
+      itemKey: "DG.Inspector.copyCaseDataToClipboard",
+      handleClick: () => {
+        if (data) {
+          if (data.collections.length > 1) {
+            copyToClipboardModal.onOpen()
+          } else {
+            navigator.clipboard.writeText(convertDatasetToCsv(data))
+            setCopiedCasesString(`${data.itemIds.length} ${data.childCollection.title}`)
+            copiedAlert.onOpen()
+          }
+        }
+      }
     },
     {
-      itemKey: "DG.Inspector.getCaseDataFromClipboard"
+      itemKey: "DG.Inspector.getCaseDataFromClipboard",
+      handleClick: () => {
```

Comment:
Can we enable/disable the menu item based on the contents of the clipboard?

Reply:
I was thinking of disabling it if there is nothing on the clipboard or if it isn't text.

Reply:
So there are two complications here.
1. Reading the clipboard is asynchronous.
2. The clipboard changing won't cause a re-render.

What I can do is run a `useEffect` on every render that reads the clipboard and then sets some state, which will take care of the first complication. The second complication seems more serious to me--if the clipboard can't be imported, we'll disable the button, and then when the user copies text to the clipboard the button will still be disabled. The opposite can happen as well. Is there a way to deal with this problem?

Reply:
It was a thought that occurred to me on review. If it's not straightforward and/or it's not the way v2 behaves feel free to punt it.